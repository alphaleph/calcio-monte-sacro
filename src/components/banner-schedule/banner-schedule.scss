@import '../../pages/globals.scss';

.banner-schedule {
    border-bottom: 1px solid $holy-yellow-team;
    padding: 4rem 0;
}

.banner-schedule-preview {
    padding-top: 1.5rem;
    padding-bottom: 4rem;
}

.banner-schedule-footer {
    padding-bottom: 2rem;
}

.banner-schedule-button:hover,
.banner-schedule-button:focus {
    color: $holy-yellow-team;
    text-decoration: underline 2px;
}