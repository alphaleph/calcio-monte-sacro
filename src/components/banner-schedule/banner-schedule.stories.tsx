import React from 'react';
import { storiesOf } from '@storybook/react';
import { TestBannerSchedule } from './banner-schedule';
import '../../pages/index.scss';

storiesOf('BannerSchedule', module).add('default', () => (
    <TestBannerSchedule />
));
