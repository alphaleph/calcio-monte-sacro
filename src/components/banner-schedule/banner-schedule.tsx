import React from 'react';
import { Link } from 'gatsby';
// import { SchedulePreview } from '../schedule-preview/schedule-preview';
import { SITE_URL } from '../../constants';
import './banner-schedule.scss';

export const BannerSchedule = () => {
    return (
        <section className="section has-background-white-main is-bold">
            <div className="container">
                <h2 className="title is-family-secondary has-text-black-main is-2">
                    SCHEDULE
                </h2>
                {/* <SchedulePreview /> */}
                <div className="columns">
                    <div className="column is-half is-offset-one-quarter">
                        <Link to={`${SITE_URL}/schedule`}>
                            <button
                                className="button is-fullwidth has-background-roma-gold has-text-black-main is-size-5 is"
                                type="button"
                            >
                                <strong>SEE FULL SCHEDULE</strong>
                            </button>
                        </Link>
                    </div>
                </div>
            </div>
        </section>
    );
};
