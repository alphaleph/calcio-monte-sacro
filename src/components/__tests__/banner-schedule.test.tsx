import React from 'react';
import { BannerSchedule } from '../banner-schedule/banner-schedule';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { mainLinkTexts } from './test-util';

describe('Banner Schedule', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<BannerSchedule />);
        expect(getByText('Schedule')).toBeInTheDocument();
        expect(getByText('See Full Schedule')).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<BannerSchedule />);
        expect(getByText('Schedule')).toBeVisible();
        expect(getByText('See Full Schedule')).toBeVisible();
        expect(getByText('See Full Schedule')).toHaveAttribute(
            'href',
            mainLinkTexts['Schedule']
        );
    });

    it('has not changed', () => {
        const { asFragment } = render(<BannerSchedule />);
        expect(asFragment()).toMatchSnapshot();
    });
});
