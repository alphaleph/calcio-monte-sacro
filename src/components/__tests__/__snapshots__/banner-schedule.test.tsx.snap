// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Schedule has not changed 1`] = `
<DocumentFragment>
  <section
    class="banner-schedule section has-background-white-main is-bold"
  >
    <div
      class="container"
    >
      <h2
        class="title is-family-secondary has-text-black-main is-2"
      >
        SCHEDULE
      </h2>
      <div
        class="banner-schedule-preview"
      >
        <section
          class="columns is-multiline"
        >
          <div
            class="column is-one-third"
          >
            <h3
              class="has-text-black-main is-size-5"
            >
              <strong>
                PREVIOUS EVENT
              </strong>
            </h3>
          </div>
          <div
            class="column is-two-thirds"
          >
            <h3
              class="has-text-black-main is-size-5"
            >
              <strong>
                UPCOMING EVENTS
              </strong>
            </h3>
          </div>
          <div
            class="column is-one-third"
          >
            <article
              class="event-card-container container"
            >
              <header
                class="event-card-header"
              >
                <p>
                  19 NOV 2019
                </p>
                <p
                  class="has-text-weight-light"
                >
                  Parco Sannazzaro
                </p>
              </header>
              <section
                class="event-card-info columns is-mobile is-vcentered has-text-centered"
              >
                <div
                  class="column is-one-third team-info"
                >
                  <p
                    aria-label="Soccer ball"
                    class="is-size-1"
                    role="img"
                  >
                    ⚽
                  </p>
                  <p>
                    CMS
                  </p>
                </div>
                <div
                  class="column is-one-third match-info has-text-team-holy"
                >
                  <p>
                    1-2
                  </p>
                </div>
                <div
                  class="column is-one-third team-info"
                >
                  <p
                    aria-label="Soccer ball"
                    class="is-size-1"
                    role="img"
                  >
                    ⚽
                  </p>
                  <p>
                    CMS2
                  </p>
                </div>
              </section>
            </article>
          </div>
          <div
            class="column is-one-third"
          >
            <article
              class="event-card-container container"
            >
              <header
                class="event-card-header"
              >
                <p>
                  19 NOV 2219
                </p>
                <p
                  class="has-text-weight-light"
                >
                  Parco Sannazzaro
                </p>
              </header>
              <section
                class="event-card-info columns is-mobile is-vcentered has-text-centered"
              >
                <div
                  class="column is-one-third team-info"
                >
                  <p
                    aria-label="Soccer ball"
                    class="is-size-1"
                    role="img"
                  >
                    ⚽
                  </p>
                  <p>
                    CMS
                  </p>
                </div>
                <div
                  class="column is-one-third match-info has-text-team-holy"
                >
                  <p>
                    21:15
                  </p>
                </div>
                <div
                  class="column is-one-third team-info"
                >
                  <p
                    aria-label="Soccer ball"
                    class="is-size-1"
                    role="img"
                  >
                    ⚽
                  </p>
                  <p>
                    CMS2
                  </p>
                </div>
              </section>
            </article>
          </div>
          <div
            class="column is-one-third"
          >
            <article
              class="event-card-container container"
            >
              <header
                class="event-card-header"
              >
                <p>
                  20 NOV 2219
                </p>
                <p
                  class="has-text-weight-light"
                >
                  Parco Sannazzaro
                </p>
              </header>
              <section
                class="event-card-info has-text-centered columns is-mobile is-vcentered"
              >
                <div
                  class="column practice-info"
                >
                  <p>
                    PRACTICE
                  </p>
                  <p
                    class="has-text-team-holy"
                  >
                    12:40
                  </p>
                </div>
              </section>
            </article>
          </div>
        </section>
      </div>
      <div
        class="columns"
      >
        <div
          class="column is-half is-offset-one-quarter"
        >
          <a
            href="https://www.calciomontesacro.com/schedule"
          >
            <button
              class="button is-fullwidth has-background-roma-gold has-text-black-main is-size-5 is"
              type="button"
            >
              <strong>
                SEE FULL SCHEDULE
              </strong>
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
